flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesTestBase.java

```diff
@@ -84,7 +85,10 @@ public void setup() throws Exception {
 		TestBaseUtils.setEnv(map);
 
 		kubeClient = server.getClient().inNamespace(NAMESPACE);
-		flinkKubeClient = new Fabric8FlinkKubeClient(flinkConfig, kubeClient);
+		flinkKubeClient = new Fabric8FlinkKubeClient(
+			flinkConfig,
+			kubeClient,
+			Executors::newDirectExecutorService);
```

Comment:
`Fabric8FlinkKubeClientTest` uses a direct executor which technically allows not to call `get()` on the pod creation/stop futures in the corresponding tests. I would still call `get()` to avoid confusion in case of any future executor refactorings

Reply:
Thanks. I will remove the direct executor and use `get` explicitly.

Reply:
I think we need to keep the direct executor in `KubernetesResourceManagerTest`. So then once the `KubernetesResourceManager` calls `createTaskManagerPod/stopPod` internally, we could do the taskmanager check immediately.

Reply:
I did not mean to remove `Executors::newDirectExecutorService` in tests, only to call `get()` in tests of `Fabric8FlinkKubeClientTest` for newly return futures.

Reply:
Got it. I will rebase the master and update this part.